noir-r1cs/src/cli/cmd/prove.rs

```diff
@@ -47,10 +51,7 @@ impl Command for Args {
         let (constraints, witnesses) = scheme.size();
         info!(constraints, witnesses, "Read Noir proof scheme");
 
-        // Read the input toml
-        let witness_file_path = &PathBuf::from(&self.witness_path);
-        let mut witness_stack = deserialize_witness_stack(witness_file_path).unwrap();
-        let witness_map: WitnessMap<NoirFieldElement> = witness_stack.pop().unwrap().witness;
+        let witness_map = execute_program_witness(&self.program_path, &self.witness_path)?;
```

Comment:
Witness computation should be part of `shceme.prove`, which should take only public and private inputs (e.g. as a toml string).